kpatch-build/create-diff-object.c

```diff
@@ -3479,6 +3548,37 @@ static void kpatch_create_mcount_sections(struct kpatch_elf *kelf)
 			rela->type = R_X86_64_PC32;
 		}
 
+#elif defined(__aarch64__)
+{
+		unsigned char *insn;
+		bool found = false;
+		int i;
+
+		if (!patchable_sec)
+			ERROR("%s: unexpected missing __patchable_function_entries section", __func__);
+
+		/* We search the patchable_sec to ensure that the symbol is patchable. */
+		list_for_each_entry(rela, &patchable_sec->relas, list) {
+			if (rela->sym && rela->sym->sec && sym->sec ==
+							   rela->sym->sec) {
+				found = true;
+				break;
+			}
```

Comment:
Why is this needed?  The check to ensure the symbol is patchable should have already been done already in kpatch_check_func_profiling_calls().  So I could be missing something, but I don't see the need to preserve `patchable_sec` or to do this check here.

I do notice that `rela->addend` is needed below for the `insn_offset` calculation, but if the offset isn't always at the same location, can the function entry point be discovered by just looking for it at the beginning of the function?  If so, I'd much rather do that because it seems less complex than the hacks needed to preserve the contents of the table.